x-pack/plugins/enterprise_search/public/applications/test_helpers/error_handling.ts

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { mockFlashMessageHelpers } from '../__mocks__/kea_logic';
+
+import { nextTick } from '@kbn/test/jest';
+import { HttpHandler } from 'src/core/public';
+
+export const itShowsServerErrorAsFlashMessage = (httpMethod: HttpHandler, callback: () => void) => {
+  const { flashAPIErrors } = mockFlashMessageHelpers;
+  it('shows any server errors as flash messages', async () => {
+    (httpMethod as jest.Mock).mockReturnValueOnce(Promise.reject('error'));
+    callback();
+    await nextTick();
+
+    expect(flashAPIErrors).toHaveBeenCalledWith('error');
+  });
+};
```

Comment:
Currently, we need to pass a specific `httpMethod` in each function call (first argument).
To simplify the call, would it be possible to mock all HTTP methods here and then do `.mockClear()` on them at the end?
That would allow us to only pass `callback` to this function during the call.